golang-sample/cloudbuild.yaml

```diff
@@ -0,0 +1,61 @@
+# Copyright 2020 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+steps:
+  # Initialize the module
+  - name: golang
+    args: ['go', 'mod', 'init', 'github.com/GoogleCloudPlatform/cloud-build-samples/golang']
+
+  # Clean up dependencies
+  - name: golang
+    args: ['go', 'mod', 'tidy']
+
+  # Build application
+  - name: golang
+    args: ['go', 'build', '.']
+
+  # Run tests and save to file
+  - name: golang
+    entrypoint: /bin/bash
+    args: 
+      - -c
+      - |
+        go test . -v -json > ${SHORT_SHA}_test_log.json
+
+  # Docker Build
+  - name: 'gcr.io/cloud-builders/docker'
+    args: ['build', '-t', 
+           'us-central1-docker.pkg.dev/$PROJECT_ID/$_REPO_NAME/myimage:$SHORT_SHA', '.']
+
+  # Docker push to Google Artifact Registry
+  - name: 'gcr.io/cloud-builders/docker'
+    args: ['push',  'us-central1-docker.pkg.dev/$PROJECT_ID/$_REPO_NAME/myimage:$SHORT_SHA']
+
+  # Deploy to Cloud Run
+  - name: google/cloud-sdk
```

Comment:
If you aren't trying to use a version of gcloud that has many of the components installed, you can save hundreds of MB in image size by using the :slim or :alpine variants. Not sure what's currently cached on the workers, but where these images are not cached it can add up to over a minute of build time.

Reply:
Tried both `gcr.io/cloud-builders/gcloud` and `googe/cloud-sdk:slim`.  The cloud-builders one runs faster b/c it says "Step #3: Already have image (with digest): gcr.io/cloud-builders/gcloud"   -- the `google/cloud-sdk:slim` image downloads the latest version. 

For this use case, we don't need the latest.  Switched to `gcr.io/cloud-builders/gcloud`